packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -141,6 +147,72 @@ export class BarrierOperatorCCAPI extends CCAPI {
 		}
 	}
 
+	public async getCapabilities(): Promise<readonly SignalType[]> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.CapabilitiesGet,
+		);
+
+		const cc = new BarrierOperatorCCCapabilitiesGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCCapabilitiesReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return response.supportedSignalTypes;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async getEvent(signalType: SignalType) {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventGet,
+		);
+
+		const cc = new BarrierOperatorCCEventGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			signalType,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCEventReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return {
+			signalType: response.type,
+			signalState: response.state,
+		};
+	}
+
+	public async setEvent(
+		signalType: SignalType,
+		signalState: SignalState,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventSet,
+		);
+
+		const cc = new BarrierOperatorCCEventSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			signalType,
+			signalState,
+		});
+		if (this.isSinglecast()) {
+			// remember the value in case the device does not respond with a target value
+			await this.getEvent(signalType);
+		}
```

Comment:
This doesn't make sense here.

Reply:
I confused myself
I think i need to move this after the sendCommand. AFter sending an event do i need to get the Event value?

Reply:
Uhh, yes. This after the sendCommand and remove the other thing.